src/service/service.py

```diff
@@ -85,20 +80,37 @@ def run(self) -> List[List[RetrievedParagraphs]]:
                     scores=[result.score for result in result],
                 )
                 doc_results.append(retrieved_documents)
-            results.append(doc_results)
-        # todo save result
-        # result_dict = responses.model_dump()
-        # self._save_result(result_dict, chunk_size=self.config.chunk_size)
+            results["retrieved_paragraphs"].append(doc_results)
+        return results
+
+    def evaluate(self, evaluation_data: dict[str, List[Any]]) -> RetrieverResults:
+        samples = evaluation_data["samples"]
+
+        retrieved_paragraphs = evaluation_data["retrieved_paragraphs"]
+        evaluator = Evaluator(self.evaluator_config)
+
+        results = evaluator.evaluate_multiple_documents(eval_samples=samples, predictions=retrieved_paragraphs)
         return results
 
-    def _save_result(self, result_dict: dict, chunk_size: int):
-        directory = self.config.out_dir + f'{current_datetime("%m%d%Y")}/'
+    def save(self, results: RetrieverResults):
+        results_dict = results.model_dump()
+
+        directory = self.evaluator_config.output_dir + f'{current_datetime("%m%d%Y")}/'
         if not os.path.exists(directory):
             os.makedirs(directory)
-        # fixme
-        filename = f'{chunk_size}_{current_datetime("%H%M%S")}.json'
-        with open(directory + filename, 'w') as ff:
-            json.dump(result_dict, ff)
+
+        os.makedirs(self.evaluator_config.output_dir, exist_ok=True)
+        data_handler_name = self.evaluator_config.data_handler.__class__.dataset_name
+        dataset_name = data_handler_name.split('/')[-1]
+
+        file_path = os.path.join(directory,
+                                 f'chunk_size_{self.splitter_config.chunk_size}_splitter_{self.splitter_config.name}'
```

Comment:
here you can just use the name of the config class

```suggestion
                                 f'chunk_size_{self.splitter_config.chunk_size}_splitter_{self.splitter_config.__class__.__name__}'
```